src/txn/txn.c

```diff
@@ -401,6 +402,24 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
 	if (!F_ISSET(txn, TXN_RUNNING))
 		WT_RET_MSG(session, EINVAL, "No transaction is active");
 
+	/*
+	 * The default sync setting is inherited from the connection, but can
+	 * be overridden by an explicit "sync" setting for this transaction.
+	 *
+	 * !!! This is an unusual use of the config code: the "default" value
+	 * we pass in is inherited from the connection.  If flush is not set in
+	 * the connection-wide flag and not overridden here, we end up clearing
+	 * all flags.
+	 */
+	WT_RET(__wt_config_gets_def(session, cfg, "sync",
+	    FLD_ISSET(txn->txn_logsync, WT_LOG_FLUSH) ? 1 : 0, &cval));
+	if (WT_STRING_MATCH("background", cval.str, cval.len))
+		txn->txn_logsync = WT_LOG_BACKGROUND;
+	else if (WT_STRING_MATCH("on", cval.str, cval.len))
+		txn->txn_logsync = S2C(session)->txn_logsync;
```

Comment:
I take the `sync=on` to be the same meaning as it was as `sync=true` which was to turn on sync using whatever method was set for `transaction_sync`.  I think we should use the method the user configured.  The assumption with overriding on the `commit_transaction` is that the user has `transaction_sync=(enabled=false)` to use in-memory buffering, but set the method to their preference so that is what is used whenever a sync is forced.